### PR TITLE
Add tests for #168 #169 and fix another edge case

### PR DESCRIPTION
**Pull request by abinoam**
cc: @carbonin

### CODE REVIEW

**Comment by abinoam** [comment]
Mr. Travis CI saved the day. Failure only on 1.9.

**Comment by abinoam** [comment]
@JEG2 could you have a look, merge and do the release. (version is already updated to 1.7.8).

File: lib/highline.rb [hunk 1]
```diff
@@ -989,12 +989,12 @@ def get_response(  )
   # instead.  This is to support any special handling for the final sequence.
   #
   def page_print( output )
-    lines = output.scan(/[^\n]*\n?/)
+    lines = output.lines.to_a
```
Comment (carbonin) [inline lib/highline.rb]: I don't think you need the `to_a` here maybe

```ruby
lines = output.split("\n")
```
would be better?

File: lib/highline.rb [hunk 2]
```diff
@@ -989,12 +989,12 @@ def get_response(  )
...
     while lines.size > @page_at
       @output.puts lines.slice!(0...@page_at).join
       @output.puts
       # Return last line if user wants to abort paging
-      return (["...\n"] + [lines.last]).join unless continue_paging?
+      return "...\n#{lines.last}" unless continue_paging?
```
Comment (carbonin) [inline lib/highline.rb]: :+1: this is definitely better this way

File: lib/highline.rb (hunk 1 above)
Comment (JEG2) [inline lib/highline.rb]: Actually, `lines.to_a` is safer.  Line endings different on different operating systems.  `\n` is typical for Unix, but Windows systems will usually have `\r\n'.  We're better off to let Ruby cover those differences for us.

**Comment by JEG2** [comment]
Released.

**Comment by abinoam** [comment]
Thank you @JEG2 and @carbonin for the comments!

File: lib/highline.rb (hunk 1 above)
Comment (abinoam) [inline lib/highline.rb]: With Ruby >= 2.0.0 String#lines returns an Array.
But when 1.9.3 is the case, String#lines returns an Enumerator.
See: 
* http://ruby-doc.org/core-2.0.0/String.html#method-i-lines
* http://ruby-doc.org/core-1.9.3/String.html#method-i-lines

That was the "why" tests were failing only on 1.9.3.
See travis log: 
* https://travis-ci.org/abinoam/highline/jobs/84543775
```
Error: test_paging(TestHighLine): NoMethodError: undefined method `size' for #<Enumerator:0x000000031c79f8>
/home/travis/build/abinoam/highline/lib/highline.rb:993:in `page_print'
/home/travis/build/abinoam/highline/lib/highline.rb:724:in `format_statement'
/home/travis/build/abinoam/highline/lib/highline.rb:621:in `say'
/home/travis/build/abinoam/highline/test/tc_highline.rb:972:in `test_paging'
```
